### PR TITLE
add baseDep to simple project generation

### PR DESCRIPTION
**Pull request by emilypi**
@ptkato

### CODE REVIEW

File: cabal-install/src/Distribution/Client/Init/FlagExtractors.hs [hunk 1]
```diff
@@ -274,6 +278,15 @@ testMainPrompt = do
...
 
+dependenciesPrompt
+    :: Interactive m
+    => InstalledPackageIndex
+    -> InitFlags
+    -> m [Dependency]
+dependenciesPrompt pkgIx flags
+  = getDependencies flags
+  $ retrieveDependencies silent flags [(fromString "Prelude", fromString "Prelude")] pkgIx
+
```
Comment (emilypi) [inline cabal-install/src/Distribution/Client/Init/FlagExtractors.hs:289]: Moved to shared prompts

**Review by ptkato — APPROVED** [review]
Everything seems to be in order.

**Review by Mikolaj — APPROVED** [review]
Low confidence approval (not due to authors, but due my lack of familiarity with the functionality).

**Comment by emilypi** [comment]
@mergifyio backport 3.6

**Pull request merged**